app/src/main/java/com/github/se/wanderpals/ui/screens/trip/finance/CategoryContent.kt

```diff
@@ -0,0 +1,207 @@
+package com.github.se.wanderpals.ui.screens.trip.finance
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Surface
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.dp
+import com.github.se.wanderpals.model.data.Category
+import com.github.se.wanderpals.model.data.Expense
+import com.github.se.wanderpals.ui.PullToRefreshLazyColumn
+
+/**
+ * Composable function to display information about expenses that have been made during the trip
+ *
+ * A pie-chart is displayed to compare the amount of expenses in each category relative to the
+ * others. The number of transaction and the total amount for each category is also displayed as
+ * items below the pie-chart.
+ *
+ * @param innerPadding Padding values for the inner content
+ * @param expenseList List of expenses
+ * @param onRefresh Callback function for handling refresh action
+ */
+@Composable
+fun CategoryContent(
+    innerPadding: PaddingValues,
+    expenseList: List<Expense>,
+    onRefresh: () -> Unit
+) {
+
+  /*Grouping expenses by category and calculating total number of transactions
+  and total amount for each category which are by default 0 if the category
+  has no associated expenses */
+  val categoryTransactionMap =
+      expenseList
+          .groupBy { it.category }
+          .mapValues { (_, expenses) ->
+            val nbPayments = expenses.size
+            val totalAmount = expenses.sumOf { it.amount }
+            Pair(nbPayments, totalAmount)
+          }
+          .withDefault { _ -> Pair(0, 0.0) }
+
+  val lazyColumn =
+      @Composable {
+        LazyColumn(
+            modifier =
+                Modifier.fillMaxWidth()
+                    .padding(innerPadding)
+                    .padding(horizontal = 25.dp, vertical = 10.dp)
```

Comment:
Weird to have 2 paddings, can be merged into a single call

Reply:
I think we can't merge it because we have padding values and horizontal/vertical padding which are not the same. However, I realized that LazyColumn field have a content padding arguments which I used to put the horizontal and vertical padding in a cleaner way.